File: ruby/ruby-sample/std/date.rb
# frozen_string_literal: true
# typed: true

require 'date'

# new date and time
date = Date.new(1999, 1, 1)
today = Date.today
datetime = DateTime.new(2008, 3, 4, 12, 0)
now = DateTime.now

# print date
p date
p datetime
puts "year:#{date.year} month:#{date.month} day:#{date.day}"
puts "hour:#{datetime.hour} minute:#{datetime.minute}, day:#{datetime.day}"

today = DateTime.now
that_day = DateTime.new(2023, 1, 1, 0, 0, 0)
difference = today - that_day
puts "2023-1-1 to today is #{difference.to_i} days"

File: python/python-sample/src/std/date.py
# %%
import datetime as dt

today = dt.date.today()
# %%
now = dt.datetime.now()

now
# %%
that_day = dt.date(1989, 10, 1)
period = today - that_day
# %%
period.days
# %%
period.days // 365
# %%
now.astimezone(dt.timezone.utc)
# %%
tz = dt.datetime.now().astimezone().tzinfo

tz
# %%
dt.datetime.now(tz=tz).strftime(r"%Y-%m-%d %H:%M:%S")

# %%
